python/cuda_cccl/cuda/compute/algorithms/_reduce.py

```
@@ -141,6 +141,29 @@ def make_reduce_into(
     Returns:
         A callable object that can be used to perform the reduction
     """
+    accum_dtype = getattr(h_init, "dtype", None)
+    if accum_dtype is None:
+        raise TypeError(
+            "Could not determine accumulator dtype from h_init; expected numpy array or object with .dtype"
+        ) from e
```

Ruff (0.15.15)

[error] Undefined name `e`

(F821)